.grit/patterns/split_trpc_router.md

```diff
@@ -13,8 +15,8 @@ pattern NamedThing($name) = or {
 }
 
 Program(body=$body) where {
-    // $prefix = "js/"
-    $filename <: r"(.+)trpcRouter.server.ts"($prefix)
+    $body <: contains `t.router($_)`
+    get_dirname($prefix)
```

Comment:
as an aside, the way predicates work to do an inner assignment is one of the things that continually confuses me. Something like `$prefix = get_dirname($body)` would feel so much more natural

Reply:
For sure, the lack of user-defined functions is a real pain point. We (overused) predicates as a hack around not having user-defined functions yet. We'll definitely prioritize user-defined functions in Marzano.